packages/shared/src/components/squads/SquadJoinButton.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { ReactElement, useMemo } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQueryClient } from 'react-query';
+import { Squad } from '../../graphql/sources';
+import { Button } from '../buttons/Button';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useToastNotification } from '../../hooks/useToastNotification';
+import { useLeaveSquad, useJoinSquad } from '../../hooks';
+
+type SquadJoinProps = {
+  className?: string;
+  squad: Squad;
+};
+
+export const SquadJoinButton = ({
+  className,
+  squad,
+}: SquadJoinProps): ReactElement => {
+  const queryClient = useQueryClient();
+  const { displayToast } = useToastNotification();
+  const { squads } = useAuthContext();
+  const isCurrentMember = useMemo(
+    () => squads?.some(({ id }) => id === squad.id) || false,
+    [squad, squads],
+  );
+
+  const { mutateAsync: joinSquad, isLoading: isJoiningSquad } = useMutation(
+    useJoinSquad({ squad }),
+    {
+      onError: () => {
+        displayToast('🚫 Something went wrong, please try again.');
```

Comment:
Please check Tom's latest PR it introduces labels for re-usable texts I see some duplicates here as well, so loved to start using it.

Reply:
Waiting for https://github.com/dailydotdev/apps/pull/1941 to merge and I'll add